### DISCUSSION TITLE
Why does Jupyter require Python to be installed? Are there any plans to not have it as a requirement in the future?

### DISCUSSION THREAD

**Discussion by EssamWisam** [jupyter/notebook#7362] [General]
As far as I know, JuPyteR stands for "Julia, Python and R". Yes, to run Julia and R one must first install Python to install Jupyter and the extension must end with `.ipynb` instead of `.ijlnb` or `.irnb` respectively. 

I previously didn't know that Jupyter requires a Python installation so this suprised me.

**Comment by jtpio** [comment]
Jupyter requires Python because Jupyter Server (serving the Notebook UI) is currently written in Python.

As for the `.ipynb` extension, this is mostly because Jupyter used to be the IPython notebook before. There might be some discussions on other repos for allowing for other extension names in the future.

**Reply by EssamWisam** [reply]
> Jupyter requires Python because Jupyter Server (serving the Notebook UI) is currently written in Python.

How about the VSCode extension? Does that require Python because as far as I know extensions are written in typescript.

I hope more extensions are available in the future (while inferring the language). I would presume that this functionality wouldn't be breaking and wouldn't be hard to support.

**Comment by jtpio** [comment]
> How about the VSCode extension? Does that require Python because as far as I know extensions are written in typescript.

The VS Code extension is likely using the same Jupyter Server, and therefore would need Python.

> I hope more extensions are available in the future (while inferring the language). I would presume that this functionality wouldn't be breaking and wouldn't be hard to support.

Linking to https://github.com/jupyter/enhancement-proposals/pull/103 as somewhat related.
